graph/telemetry/istio/appender/idle_node.go

```diff
@@ -31,7 +31,7 @@ func (a IdleNodeAppender) AppendGraph(trafficMap graph.TrafficMap, globalInfo *g
 
 	services := []models.ServiceDetails{}
 	workloads := []models.WorkloadListItem{}
-	clusterName := graph.Unknown
+	clusterName := globalInfo.HomeCluster // This can be empty if another appender hasn't populated this yet.
```

Comment:
Don't worry about just setting globalInfo.HomeCluster immediately.   Look at dead_node.go#AppendGraph and take that same code an plunk it here.  Then you can assume it's set.

Reply:
The dead node appender simply ignores errors if they happen: https://github.com/kiali/kiali/blob/master/graph/telemetry/istio/appender/dead_node.go#L31-L36 whereas the idle node implementation panics if any errors occur: https://github.com/kiali/kiali/blob/master/graph/telemetry/istio/appender/idle_node.go#L45. Which is the correct way we want to handle errors in this case?

Reply:
hmm, good catch, not sure why dead_node is ignoring an error there.  It should be using graph.CheckError(err).  Feel free to fix that as part of this PR.

Reply:
I tried [adding a test for the check/panic](https://github.com/nrfox/kiali/compare/kiali-3917...nrfox:debug-test?expand=1#diff-5a6963c484616b4837918ea3b52565f7ca434a7aa00d4f3a94be6965f31b3836R211-R229) and got pretty close but if another test ran before that one the cached value of cluster info would get returned causing the test to pass/fail based on the order of the test runs. I couldn't figure out a way to edit that global cache value so if you've got any suggestions there I'm happy to add the test back in.

Reply:
don't worry about the panic test.